edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/PrometheusMetricsParser.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Newtonsoft.Json;
+
+    public static class PrometheusMetricsParser
+    {
+        // Extracts data from prometheous format. Read about prometheous format here:
+        // https://prometheus.io/docs/concepts/data_model/
+        const string PrometheusMetricSchema = @"^(?<metricname>[^#\{\}]+)(\{((?<tagname>[^="",]+)=(\""(?<tagvalue>[^="",]+)\"")(,(?<tagname>[^="",]+)=(\""(?<tagvalue>[^="",]+)\""))*)\})?\s(?<metricvalue>.+)$";
+        static readonly Regex PrometheusSchemaRegex = new Regex(PrometheusMetricSchema, RegexOptions.Compiled);
+
+        public static IEnumerable<Metric> ParseMetrics(DateTime timeGeneratedUtc, string prometheusMessage)
+        {
+            Preconditions.CheckArgument(timeGeneratedUtc.Kind == DateTimeKind.Utc, $"Metric {nameof(timeGeneratedUtc)} parameter only supports in UTC.");
+            using (StringReader sr = new StringReader(prometheusMessage))
+            {
+                string line;
+                while ((line = sr.ReadLine()) != null)
```

Comment:
Remove other Trim method call.
```suggestion
                while ((line = sr.ReadLine().Trim()) != null)
```